tests/test_sims_corr.py

```diff
@@ -0,0 +1,211 @@
+import unittest
+from graspy.simulations.simulations_corr import sample_edges_corr, er_corr, sbm_corr
+import numpy as np
+import pytest
+import warnings
+
+
+class Test_Sample_Corr(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.n = 500
+        cls.p = 0.5
+        cls.r = 0.3
+        cls.P = cls.p * np.ones((cls.n, cls.n))
+        cls.R = cls.r * np.ones((cls.n, cls.n))
+
+    def test_bad_input(self):
+        with self.assertRaises(TypeError):
+            P = 10
+            sample_edges_corr(P, self.R, directed=False, loops=False)
+
+        with self.assertRaises(TypeError):
+            P = "0.5"
+            sample_edges_corr(P, self.R, directed=False, loops=False)
+
+        with self.assertRaises(TypeError):
+            R = 10
+            sample_edges_corr(self.P, R, directed=False, loops=False)
+
+        with self.assertRaises(TypeError):
+            R = "0.5"
+            sample_edges_corr(self.P, R, directed=False, loops=False)
+
+        with pytest.raises(TypeError):
+            sample_edges_corr(self.P, self.r, directed="hey", loops=False)
+
+        with pytest.raises(TypeError):
+            sample_edges_corr(self.P, self.r, directed=False, loops=6)
+
+    def test_sample_edges_corr(self):
+        # P = self.p * np.ones((self.n, self.n))
+        g1, g2 = sample_edges_corr(self.P, self.R, directed=False, loops=False)
+
+        # check the 1 probability of the output binary matrix
+        # should be near to the input P matrix
+        self.assertTrue(
+            np.isclose(self.p, g2.sum() / (self.n * (self.n - 1)), atol=0.05)
+        )
+
+        # check rho
+        k1 = g1.copy()
+        k2 = g2.copy()
+        k1 = k1[np.where(~np.eye(k1.shape[0], dtype=bool))]
+        k2 = k2[np.where(~np.eye(k2.shape[0], dtype=bool))]
+        output_r = np.corrcoef(k1, k2)[0, 1]
+        self.assertTrue(np.isclose(self.r, output_r, atol=0.06))
+
+        # check the similarity of g1 and g2
```

Comment:
is this redundant with checking the correlation?

Reply:
i know you showed me at some point but i still don't really get what this is

Reply:
I can delete it if you think it is unnecessary. I just want to see how close the graph pair is.